trucks.js

```diff
@@ -1,254 +1,268 @@
+        schedule: ["Sunday"]
+    },
+    {
+        name: "BUNS",
+        type: "Burgers",
+        schedule: ["Tuesday", "Wednesday", "Friday"]
+    },
+    {
+        name: "Cake Mobile",
+        type: "Sweets",
+        schedule: []
+    },
+    {
+        name: "Caravan Crepes",
+        type: "French",
+        schedule: ["Wednesday", "Thursday", "Friday"]
+    },
+    {
+        name: "Charlie's",
+        type: "Burgers/Cheesesteaks",
+        schedule: ["Monday", "Tuesday", "Wednesday","Thursday"]
+    },
+    {
+        name: "Cheese Wizards",
+        type: "Sandwiches",
+        schedule: ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday"]
+    },
+    {
+        name: "Chewaya Authentic Moroccan Sanwiches",
+        type: "Moroccan",
+        schedule: []
+    },
+    {
+        name: "Chick'n Fix",
+        type: "American/Fillipino",
+        schedule: ["Monday", "Tuesday", "Wednesday", "Thursday"]
+    },
+    {
+        name: "Chopstix",
+        type: "Asian Fusion",
+        schedule: ["Tuesday", "Wednesday", "Thursday"]
+    },
+    {
+        name: "Contigo",
+        type: "Mexican",
+        schedule: []
+    },
+    {
+        name: "Convoy Coffee",
+        type: "Coffee",
+        schedule: ["Wednesday", "Saturday", "Sunday"]
+    },
+    {
+        name: "Crisp Creperie",
+        type: "French",
+        schedule: ["Tuesday"]
+    },
+    {
+        name: "Curb Jumper Street Eats",
+        type: "Sandwiches/Burgers",
+        schedule: ["Monday", "Wednesday"]
+    },
+    {
+        name: "Curbside",
+        type: "Vietnamese",
+        schedule: []
+    },
+    {
+        name: "Danielle's Crepes",
+        type: "French",
+        schedule: ["Tuesday", "Friday"]
+    },
+    {
+        name: "Dante's Inferno Dogs",
+        type: "Hot Dogs",
+        schedule: []
+    },
+    {
+        name: "Delicatessen Montanti",
+        type: "Sandwiches",
+        schedule: ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday"]
+    },
+    {
+        name: "Diablo Food Truckz",
+        type: "Asian",
+        schedule: []
+    },
+    {
+        name: "Djung on Wheels",
+        type: "Thai",
+        schedule: ["Monday", "Tuesday", "Wednesday", "Friday"]
+    },
+    {
+        name: "Dog Japon",
+        type: "Hot Dogs",
+        schedule: ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"]
+    },
+    {
+        name: "Dogfather Catering",
+        type: "Hot dogs, sausages, wood fired pizza",
+        schedule: []
+    },
+    {
+        name: "Don Lucho's",
+        type: "Peruvian Sandwiches",
+        schedule: []
+    }
 ];
 
-/* 
+/*
  * return an object from the module with a single method on it: filterByDay
  * that method should take a single parameter that represents the day to filter on
  * use the built-in filter() method to return all trucks that have the day in their
  * schedule
  */
+
+function filterTrucksByDay(day) {
+
+    function filterByDay(element) {
```

Comment:
While this works fine, I would avoid declaring a function inside a function like this. In general that makes code more reusable and more testable.